Resources/config/services.xml

```diff
@@ -68,5 +68,6 @@
         <service id="httplug.factory.socket" class="Http\HttplugBundle\ClientFactory\SocketFactory" public="false">
             <argument type="service" id="httplug.message_factory"/>
         </service>
+        <service id="httplug.factory.mock" class="Http\HttplugBundle\ClientFactory\MockFactory" public="false" />
```

Comment:
i think this service should be removed from the container when the mock client is not available. that way, users notice problems at container build time instead of at runtime.